app/main.py

```diff
@@ -1,3 +1,74 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance" | int | float) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
```

Comment:
Not  really necessary to return NotImplemented here and in other methods. If you don't use it then it can simplify your code and make it more readable and contain less instructions. For example:
```
    def __add__(self, other: Distance | int) -> Distance:
        if isinstance(other, Distance):
            return Distance(self.km + other.km)
        return Distance(self.km + other)
Here you have only three lines of code and only one if condition. And then if data type is incorrect and adding could not happen an exception will be raised which is perfectly fine in this scenario